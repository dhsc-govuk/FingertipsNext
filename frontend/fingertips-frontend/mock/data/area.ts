export const mockAreaData = (id: string) => ({
  id,
  name: `Greater Manchester ICB - ${id} sub-location`,
  groupType: `GT-${id}`,
  group: 'QOP',
});

export const mockAvailableGroupTypes = [
  {
    id: 'GT-001',
    name: 'Integrated Care Board',
  },
  {
    id: 'GT-002',
    name: 'Integrated Care Panel',
  },
  {
    id: 'GT-003',
    name: 'Integrated Care Shelf',
  },
];

export const mockAvailableGroup = [
  {
    id: 'QOP',
    name: 'Greater Manchester  ICB - QOP',
  },
  {
    id: 'LOP',
    name: 'Leeds  ICB - LOP',
  },
  {
    id: 'SOP',
    name: 'Sheffield  ICB - SOP',
  },
];

export const mockAvailableAreasInGroup = [
  {
    id: '001',
    name: 'Greater Manchester ICN - 001',
  },
  {
    id: '002',
    name: 'Greater Manchester ICN - 002',
  },
  {
    id: '003',
    name: 'Greater Manchester ICN - 003',
  },
  {
    id: '004',
    name: 'Greater Manchester ICN - 004',
  },
  {
    id: '005',
    name: 'Greater Manchester ICN - 005',
  },
  {
    id: '006',
    name: 'Greater Manchester ICN - 006',
  },
  {
    id: '007',
    name: 'Greater Manchester ICN - 007',
  },
  {
    id: '008',
    name: 'Greater Manchester ICN - 008',
  },
  {
    id: '009',
    name: 'Greater Manchester ICN - 008',
  },
  {
    id: '010',
    name: 'Greater Manchester ICN - 010',
  },
];
