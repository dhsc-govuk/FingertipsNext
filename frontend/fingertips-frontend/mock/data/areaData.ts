import { AreaType, AreaWithRelations } from '@/generated-sources/ft-api-client';

export const mockAreaTypes: AreaType[] = [
  {
    name: 'Counties & UAs',
    level: 2,
    hierarchyName: 'Admin',
  },
  {
    name: 'Country',
    level: 0,
    hierarchyName: 'All',
  },
  {
    name: 'GP',
    level: 4,
    hierarchyName: 'NHS',
  },
  {
    name: 'ICB',
    level: 2,
    hierarchyName: 'NHS',
  },
  {
    name: 'NHS region',
    level: 1,
    hierarchyName: 'NHS',
  },
  {
    name: 'PCN',
    level: 3,
    hierarchyName: 'NHS',
  },
  {
    name: 'Regions Statistical',
    level: 1,
    hierarchyName: 'Admin',
  },
];

export const mockAreaDataForCountry: Record<string, AreaWithRelations> = {
  E92000001: {
    code: 'E92000001',
    name: 'England',
    hierarchyName: 'All',
    areaType: 'Country',
    level: 0,
  },
};

export const mockAreaDataForRegionsStatistical: Record<
  string,
  AreaWithRelations
> = {
  E12000001: {
    code: 'E12000001',
    name: 'North East region (statistical)',
    hierarchyName: 'Admin',
    areaType: 'Regions Statistical',
    level: 1,
  },
  E12000002: {
    code: 'E12000002',
    name: 'North West region (statistical)',
    hierarchyName: 'Admin',
    areaType: 'Regions Statistical',
    level: 1,
  },
  E12000003: {
    code: 'E12000003',
    name: 'Yorkshire and the Humber region (statistical)',
    hierarchyName: 'Admin',
    areaType: 'Regions Statistical',
    level: 1,
  },
};

export const mockAreaDataForNHSRegion: Record<string, AreaWithRelations> = {
  E40000007: {
    code: 'E40000007',
    name: 'East of England NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000003: {
    code: 'E40000003',
    name: 'London NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000005: {
    code: 'E40000005',
    name: 'South East NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000006: {
    code: 'E40000006',
    name: 'South West NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000010: {
    code: 'E40000010',
    name: 'North West NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000011: {
    code: 'E40000011',
    name: 'Midlands NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000012: {
    code: 'E40000012',
    name: 'North East and Yorkshire NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
};

export const mockAreaDataForCountiesAndUAs: Record<string, AreaWithRelations> =
  {
    E06000047: {
      code: 'E06000047',
      name: 'County Durham',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
    E06000005: {
      code: 'E06000005',
      name: 'Darlington',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
    E08000037: {
      code: 'E08000037',
      name: 'Gateshead',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
  };

export const mockAreaDataForICB: Record<string, AreaWithRelations> = {
  E38000007: {
    code: 'E38000007',
    name: 'NHS Basildon And Brentwood ICB',
    hierarchyName: 'NHS',
    areaType: 'ICB',
    level: 2,
  },
  E38000026: {
    code: 'E38000026',
    name: 'NHS Cambridgeshire and Peterborough ICB',
    hierarchyName: 'NHS',
    areaType: 'ICB',
    level: 2,
  },
  E38000240: {
    code: 'E38000240',
    name: 'NHS North Central London ICB',
    hierarchyName: 'NHS',
    areaType: 'ICB',
    level: 2,
  },
  E38000244: {
    code: 'E38000244',
    name: 'NHS South East London ICB',
    hierarchyName: 'NHS',
    areaType: 'ICB',
    level: 2,
  },
};

export const mockAreaDataForPCN: Record<string, AreaWithRelations> = {
  U15488: {
    code: 'U15488',
    name: 'East Basildon PCN',
    hierarchyName: 'NHS',
    areaType: 'PCN',
    level: 3,
  },
  U55146: {
    code: 'U55146',
    name: 'Central Basildon PCN',
    hierarchyName: 'NHS',
    areaType: 'PCN',
    level: 3,
  },
  U02795: {
    code: 'U02795',
    name: 'North 2 Islington PCN',
    hierarchyName: 'NHS',
    areaType: 'PCN',
    level: 3,
  },
  U05885: {
    code: 'U05885',
    name: 'South Camden PCN',
    hierarchyName: 'NHS',
    areaType: 'PCN',
    level: 3,
  },
};

export const mockAreaDataForGP: Record<string, AreaWithRelations> = {
  F81186: {
    code: 'F81186',
    name: 'Felmores Medical Centre',
    hierarchyName: 'NHS',
    areaType: 'GP',
    level: 4,
  },
  F81640: {
    code: 'F81640',
    name: 'Aryan Medical Centre',
    hierarchyName: 'NHS',
    areaType: 'GP',
    level: 4,
  },
  F83004: {
    code: 'F83004',
    name: 'Archway Medical Centre',
    hierarchyName: 'NHS',
    areaType: 'GP',
    level: 4,
  },
  F83008: {
    code: 'F83008',
    name: 'The Goodinge Group Practice',
    hierarchyName: 'NHS',
    areaType: 'GP',
    level: 4,
  },
};

export const mockAreaData: Record<string, AreaWithRelations> = {
  ...mockAreaDataForCountry,
  ...mockAreaDataForNHSRegion,
  ...mockAreaDataForRegionsStatistical,
  ...mockAreaDataForCountiesAndUAs,
  ...mockAreaDataForICB,
  ...mockAreaDataForPCN,
  ...mockAreaDataForGP,
};

export const mockAvailableAreas: Record<string, Area[]> = {
  'Counties & UAs': [
    {
      code: 'E06000047',
      name: 'County Durham',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
    {
      code: 'E06000005',
      name: 'Darlington',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
    {
      code: 'E08000037',
      name: 'Gateshead',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
  ],
  'Country': [
    {
      code: 'E92000001',
      name: 'England',
      hierarchyName: 'All',
      areaType: 'Country',
      level: 0,
    },
  ],
  'GP': [
    {
      code: 'F81186',
      name: 'Felmores Medical Centre',
      hierarchyName: 'NHS',
      areaType: 'GP',
      level: 4,
    },
    {
      code: 'F81640',
      name: 'Aryan Medical Centre',
      hierarchyName: 'NHS',
      areaType: 'GP',
      level: 4,
    },
    {
      code: 'F83004',
      name: 'Archway Medical Centre',
      hierarchyName: 'NHS',
      areaType: 'GP',
      level: 4,
    },
    {
      code: 'F83008',
      name: 'The Goodinge Group Practice',
      hierarchyName: 'NHS',
      areaType: 'GP',
      level: 4,
    },
  ],
  'ICB': [
    {
      code: 'E38000007',
      name: 'NHS Basildon And Brentwood ICB',
      hierarchyName: 'NHS',
      areaType: 'ICB',
      level: 2,
    },
    {
      code: 'E38000026',
      name: 'NHS Cambridgeshire and Peterborough ICB',
      hierarchyName: 'NHS',
      areaType: 'ICB',
      level: 2,
    },
    {
      code: 'E38000240',
      name: 'NHS North Central London ICB',
      hierarchyName: 'NHS',
      areaType: 'ICB',
      level: 2,
    },
    {
      code: 'E38000244',
      name: 'NHS South East London ICB',
      hierarchyName: 'NHS',
      areaType: 'ICB',
      level: 2,
    },
  ],
  'NHS region': [
    {
      code: 'E40000007',
      name: 'East of England NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000003',
      name: 'London NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000005',
      name: 'South East NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000006',
      name: 'South West NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000010',
      name: 'North West NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000011',
      name: 'Midlands NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000012',
      name: 'North East and Yorkshire NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
  ],
  'PCN': [
    {
      code: 'U15488',
      name: 'East Basildon PCN',
      hierarchyName: 'NHS',
      areaType: 'PCN',
      level: 3,
    },
    {
      code: 'U55146',
      name: 'Central Basildon PCN',
      hierarchyName: 'NHS',
      areaType: 'PCN',
      level: 3,
    },
    {
      code: 'U02795',
      name: 'North 2 Islington PCN',
      hierarchyName: 'NHS',
      areaType: 'PCN',
      level: 3,
    },
    {
      code: 'U05885',
      name: 'South Camden PCN',
      hierarchyName: 'NHS',
      areaType: 'PCN',
      level: 3,
    },
  ],
  'Regions Statistical': [
    {
      code: 'E12000001',
      name: 'North East region (statistical)',
      hierarchyName: 'Admin',
      areaType: 'Regions Statistical',
      level: 1,
    },
    {
      code: 'E12000002',
      name: 'North West region (statistical)',
      hierarchyName: 'Admin',
      areaType: 'Regions Statistical',
      level: 1,
    },
    {
      code: 'E12000003',
      name: 'Yorkshire and the Humber region (statistical)',
      hierarchyName: 'Admin',
      areaType: 'Regions Statistical',
      level: 1,
    },
  ],
};
