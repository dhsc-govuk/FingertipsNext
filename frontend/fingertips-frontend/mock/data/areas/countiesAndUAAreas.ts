import { Area } from '@/generated-sources/ft-api-client';
import { countiesAndUnitaryAuthoritiesAreaType } from '../../../lib/areaFilterHelpers/areaType';

export const countyDurham: Area = {
  code: 'E06000047',
  name: 'County Durham',
  areaType: countiesAndUnitaryAuthoritiesAreaType,
};

export const darlington: Area = {
  code: 'E06000005',
  name: 'Darlington',
  areaType: countiesAndUnitaryAuthoritiesAreaType,
};

export const gateshead: Area = {
  code: 'E08000037',
  name: 'Gateshead',
  areaType: countiesAndUnitaryAuthoritiesAreaType,
};
