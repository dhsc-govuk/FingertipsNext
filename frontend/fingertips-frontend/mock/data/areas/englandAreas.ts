import { Area } from '@/generated-sources/ft-api-client';
import { englandAreaType } from '../areaType';

export const englandArea: Area = {
  code: 'E92000001',
  name: 'England',
  areaType: englandAreaType,
};
