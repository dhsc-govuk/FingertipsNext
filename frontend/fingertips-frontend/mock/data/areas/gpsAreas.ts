import { Area } from '@/generated-sources/ft-api-client';
import { gpsAreaType } from '../areaType';

export const felmoresGP: Area = {
  code: 'F81186',
  name: 'Felmores Medical Centre',
  areaType: gpsAreaType,
};

export const aryanGP: Area = {
  code: 'F81640',
  name: 'Aryan Medical Centre',
  areaType: gpsAreaType,
};

export const archwayGP: Area = {
  code: 'F83004',
  name: 'Archway Medical Centre',
  areaType: gpsAreaType,
};

export const goodingeGP: Area = {
  code: 'F83008',
  name: 'The Goodinge Group Practice',
  areaType: gpsAreaType,
};
