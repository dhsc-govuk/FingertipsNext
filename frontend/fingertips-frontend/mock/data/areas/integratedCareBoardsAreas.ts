import { Area } from '@/generated-sources/ft-api-client';
import { nhsIntegratedCareBoardsAreaType } from '../../../lib/areaFilterHelpers/areaType';

export const basildonAndBrentwoodICB: Area = {
  code: 'E38000007',
  name: 'NHS Basildon And Brentwood ICB',
  areaType: nhsIntegratedCareBoardsAreaType,
};

export const cambridgeAndPeterboroughICB: Area = {
  code: 'E38000026',
  name: 'NHS Cambridgeshire and Peterborough ICB',
  areaType: nhsIntegratedCareBoardsAreaType,
};

export const northCentralLondonICB: Area = {
  code: 'E38000240',
  name: 'NHS North Central London ICB',
  areaType: nhsIntegratedCareBoardsAreaType,
};

export const southEastLondonICB: Area = {
  code: 'E38000244',
  name: 'NHS South East London ICB',
  areaType: nhsIntegratedCareBoardsAreaType,
};
