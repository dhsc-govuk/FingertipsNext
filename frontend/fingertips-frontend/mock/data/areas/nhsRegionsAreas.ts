import { Area } from '@/generated-sources/ft-api-client';
import { nhsRegionsAreaType } from '../../../lib/areaFilterHelpers/areaType';

export const eastEnglandNHSRegion: Area = {
  code: 'E40000007',
  name: 'East of England NHS Region',
  areaType: nhsRegionsAreaType,
};

export const londonNHSRegion: Area = {
  code: 'E40000003',
  name: 'London NHS Region',
  areaType: nhsRegionsAreaType,
};

export const southEastNHSRegion: Area = {
  code: 'E40000005',
  name: 'South East NHS Region',
  areaType: nhsRegionsAreaType,
};

export const southWestNHSRegion: Area = {
  code: 'E40000006',
  name: 'South West NHS Region',
  areaType: nhsRegionsAreaType,
};

export const northWestNHSRegion: Area = {
  code: 'E40000010',
  name: 'North West NHS Region',
  areaType: nhsRegionsAreaType,
};

export const midlandsNHSRegion: Area = {
  code: 'E40000011',
  name: 'Midlands NHS Region',
  areaType: nhsRegionsAreaType,
};

export const northEastAndYorkshireNHSRegion: Area = {
  code: 'E40000012',
  name: 'North East and Yorkshire NHS Region',
  areaType: nhsRegionsAreaType,
};
