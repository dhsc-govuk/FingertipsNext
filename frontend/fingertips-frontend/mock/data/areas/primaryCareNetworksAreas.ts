import { Area } from '@/generated-sources/ft-api-client';
import { nhsPrimaryCareNetworksAreaType } from '../../../lib/areaFilterHelpers/areaType';

export const eastBasildonPCN: Area = {
  code: 'U15488',
  name: 'East Basildon PCN',
  areaType: nhsPrimaryCareNetworksAreaType,
};

export const centralBasildonPCN: Area = {
  code: 'U55146',
  name: 'Central Basildon PCN',
  areaType: nhsPrimaryCareNetworksAreaType,
};

export const north2IslingtonPCN: Area = {
  code: 'U02795',
  name: 'North 2 Islington PCN',
  areaType: nhsPrimaryCareNetworksAreaType,
};

export const southCamdenPCN: Area = {
  code: 'U05885',
  name: 'South Camden PCN',
  areaType: nhsPrimaryCareNetworksAreaType,
};
