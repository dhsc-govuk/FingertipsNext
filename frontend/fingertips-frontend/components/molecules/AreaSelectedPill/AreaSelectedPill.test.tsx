import { render, screen } from '@testing-library/react';
import { AreaSelectedPill } from './index';
import { userEvent } from '@testing-library/user-event';
import { mockAreaDataForNHSRegion } from '@/mock/data/areaData';
import { eastEnglandNHSRegion } from '@/mock/data/areas/nhsRegionsAreas';

describe('AreaSelectedPill', () => {
  const mockRemoveFilter = vi.fn();
  const area = mockAreaDataForNHSRegion[eastEnglandNHSRegion.code];

  beforeEach(() => {
    mockRemoveFilter.mockClear();
  });

  it('should render the AreaSelectedPill correctly', () => {
    render(<AreaSelectedPill area={area} onRemoveFilter={mockRemoveFilter} />);
    expect(screen.getByText(area.name)).toBeInTheDocument();
    expect(screen.getByText(area.areaType.name)).toBeInTheDocument();
  });

  it('should call onRemoveFilter when the pill is clicked', async () => {
    render(<AreaSelectedPill area={area} onRemoveFilter={mockRemoveFilter} />);
    const user = userEvent.setup();
    await user.click(screen.getByTestId('remove-icon-div'));
    expect(mockRemoveFilter).toHaveBeenCalledTimes(1);
    expect(mockRemoveFilter).toHaveBeenCalledWith(area.code);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <AreaSelectedPill area={area} onRemoveFilter={mockRemoveFilter} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
