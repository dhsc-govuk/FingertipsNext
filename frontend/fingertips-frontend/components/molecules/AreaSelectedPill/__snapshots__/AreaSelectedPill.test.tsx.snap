// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AreaSelectedPill > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gEvEer cmUCgj"
    data-testid="pill-container"
  >
    <button
      aria-label="Remove area East of England NHS Region"
      class="sc-fqkvVR dcQCKA"
      data-testid="remove-icon-div"
    >
      <svg
        class="sc-aXZVg kKRHvy"
        data-testid="x-icon"
        fill="none"
        height="16"
        stroke="#000000"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="3"
        viewBox="0 0 25 25"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </button>
    <div
      class="sc-eqUAAy eqmima"
      data-testid="filter-name"
    >
      <p
        style="margin: 0px;"
      >
        <span
          class="sc-dcJsrY gzjaQs"
        >
          East of England NHS Region
        </span>
         
        <span
          class="sc-iGgWBj iaVgID"
        >
          NHS Regions
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
